### PR TITLE
happy family with Makefile and sonar properties

### PR DESCRIPTION
**Pull request by chunchet**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,38 +1,40 @@
...
-    cd build/release && $(MAKE)
+	cd build/release && $(MAKE)
 
 test:
-    cd build/debug && \
-    ./unittest --gtest_output=xml:../../gtest_unittest.xml
+	cd build/debug && \
+	./unittest --gtest_output=xml:../../gtest_unittest.xml
+
+test-algorithm:
```
Comment (flyingbag) [inline Makefile]: You don't need this.

**Comment by flyingbag** [comment]
*.swp, *.xml files are not supposed to be there. Please remove them.

File: Makefile [hunk 2]
```diff
@@ -1,38 +1,40 @@
 default: cmake-debug debug
-jenkins: clean cmake-debug debug test coverage cmake-release release lint chown
+jenkins: clean cmake-debug cmake-release debug release test test-algorithm coverage lint
 
 cmake-debug:
-    mkdir -p build/debug && \
-    cd build/debug && \
-    cmake -DCMAKE_BUILD_TYPE=Debug ../../
+	mkdir -p build/debug && \
```
Comment (flyingbag) [inline Makefile]: Use spaces instead of tab.

**Comment by chunchet** [comment]
Changes made:
- remove test-algorithm from Makefile
- remove .*.swp and *.xml files

**Pull request merged**